web/renderer/components/pageComponents/DatabasePage/ForRemotes/RemotesPage/DoltLoginButton.tsx

```diff
@@ -0,0 +1,124 @@
+import { Button, ErrorMsg, QueryHandler } from "@dolthub/react-components";
+import {
+  DatabaseConnectionFragment,
+  useCurrentConnectionQuery,
+} from "@gen/graphql-types";
+import { useState } from "react";
+import { useDelay } from "@dolthub/react-hooks";
+import css from "./index.module.css";
+
+type InnerProps = {
+  connection?: DatabaseConnectionFragment | null;
+};
+
+type LoggedInUser = {
+  email: string;
+  username: string;
+};
+
+function Inner({ connection }: InnerProps) {
+  const [err, setErr] = useState<Error | undefined>(undefined);
+  const [loggedInUser, setLoggedInUser] = useState<LoggedInUser | undefined>(
+    undefined,
+  );
+  const [pending, setPending] = useState(false);
+  const [cancelId, setCancelId] = useState<string | null>(null);
+  const success = useDelay(3000);
+
+  if (!connection?.isLocalDolt) return null;
+
+  const onLogin = async () => {
+    let requestId: string | null = null;
+
+    try {
+      setPending(true);
+      setErr(undefined);
+
+      // Set up the listener for login process id
+      const idPromise = new Promise<string>(resolve => {
+        window.ipc.onLoginStarted((receivedId: string) => {
+          resolve(receivedId);
+        });
+      });
+
+      // Start login process
+      const loginPromise = window.ipc.doltLogin(connection.name);
+
+      // Wait for either the ID or login completion
+      requestId = await Promise.race([
+        idPromise.then(id => {
+          setCancelId(id);
+        }),
+        loginPromise.then(() => null),
+      ]);
+
+      const result = await loginPromise;
+      if (result.success) {
+        setLoggedInUser({
+          email: result.email,
+          username: result.username,
+        });
+        success.start();
+      }
+    } catch (error) {
+      setErr(new Error(`Failed to login: ${error}`));
+    } finally {
+      setPending(false);
+      // Cleanup ID reference
+      if (requestId) setCancelId(null);
```

Comment:
Can this go at the end of the `try` block instead? Then you don't need to reassign the `requestId` variable